src/functions.php

```diff
@@ -231,6 +232,19 @@ function ($reason, $idx, Promise $aggregate) {
         ksort($results);
         return $results;
     });
+
+    if (true === $recursive) {
+        $promise = $promise->then(function ($results) use ($recursive, &$promises) {
+            foreach ($promises AS $promise) {
+                if (\GuzzleHttp\Promise\PromiseInterface::PENDING === $promise->getState()) {
+                    return all($promises, $recursive);
```

Comment:
That's a good point.

I oriented my tests on wait functions that return a fixed value, and assumed the already settled promises won't have their wait function called anymore, since they should be already resolved or rejected. I have to check this in a deeper way. Do not merge for the moment.

If I was wrong, what do you suggest to loop over the new pending promises? At first I thought putting them in a new `ArrayIterator` but performance impact apart, this would force the original `$promises` type hint and if the user defined a `MyPromisesStackIterator` with an `addNewPromise()` method, for instance, he would not be able to call that new `ArrayIterator::addNewPromise()`.

That's why I was attached in keeping the original `iterable` passed to the function.